Lib/subprocess.py

```diff
@@ -955,15 +956,22 @@ def __init__(self, args, bufsize=-1, executable=None,
                     self.stderr = io.TextIOWrapper(self.stderr,
                             encoding=encoding, errors=errors)
 
-            self._execute_child(args, executable, preexec_fn, close_fds,
-                                pass_fds, cwd, env,
-                                startupinfo, creationflags, shell,
-                                p2cread, p2cwrite,
-                                c2pread, c2pwrite,
-                                errread, errwrite,
-                                restore_signals,
-                                gid, gids, uid, umask,
-                                start_new_session)
+            try:
+                self._execute_child(args, executable, preexec_fn, close_fds,
+                                    pass_fds, cwd, env,
+                                    startupinfo, creationflags, shell,
+                                    p2cread, p2cwrite,
+                                    c2pread, c2pwrite,
+                                    errread, errwrite,
+                                    restore_signals,
+                                    gid, gids, uid, umask,
+                                    start_new_session)
+            except OSError as exc:
+                if exec_raise:
+                    raise
+                self.returncode = exc.errno
```

Comment:
I think it is dangerous to assign errno as returncode as someone will wind up depending on that behavior even though they have no way of knowing if it was an errno or an actual exit code of the process.

The user said they didn't want an exception with full information, so lets not muck things up by potentially clobbering other information they could've had.  Return a static value out of range of any possible actual child return code on all platforms, or if we want to get this complicated at all: Let the user specify the value to use as returncode.

Agreed that exec_raise and exec_* aren't great names for the parameter.  `oserror_via_returncode` perhaps as a way to be explicit?  But how many users even think in terms of OSError anyways given that is a parent class of the actual (FileNotFoundError, NotADirectoryError, PermissionError)'s they're likely to encounter?